File: exercism/ecmascript/word-count/word-count.js
const regex = /[\s* \s*]/;

class Words {
  count(sentance){
    return sentance
      .toLowerCase()
      .split( regex )
      .filter( word => word )
      .reduce( (acc, word) => {
        !Number.isInteger(acc[word])
          ? acc[word] = 1
          : acc[word] += 1
        return acc
      },{})
  }
}

export default Words

File: elixir/exercism/elixir/word-count/lib/word_count.ex
defmodule WordCount do
  @doc """
  Count the number of words in the sentence.

  Words are compared case-insensitively.
  """
  @spec count(String.t()) :: map
  def count(sentence) do
    cleaned = Regex.replace(~r/[^\p{L}\p{Nd}\-]+/u, sentence, " ")

    cleaned
    |> String.downcase()
    |> String.split(" ")
    |> Enum.filter(fn item -> item !== "" end)
    |> Enum.reduce(%{}, fn word, acc ->
      Map.merge(acc, %{word => 1}, fn _k, v1, v2 ->
        v1 + v2
      end)
    end)
  end
end
